src/CSET/operators/plot.py

```diff
@@ -931,6 +950,68 @@ def _spatial_plot(
     _make_plot_html_page(complete_plot_index)
 
 
+def _convert_precipitation_units_callback(cube: iris.cube.Cube):
+    """To convert the unit of precipitation from kg m-2 s-1 to mm hr-1.
+
+    Some precipitation diagnostics are output with unit kg m-2 s-1 and are converted to mm hr-1.
+    """
+    # if cube.attributes["STASH"] == "m01s04i203" or cube.long_name == "surface_microphysical_rainfall_rate":
+    if cube.long_name == "surface_microphysical_rainfall_rate":
+        if cube.units == "kg m-2 s-1":
+            logging.info("Converting precipitation units from kg m-2 s-1 to mm hr-1")
+            # manually convert from kg m-2 s-1 to mm hr-1 assuming 1kg water = 1l water = 1dm^3 water
+            cube.data = cube.data * 3600.0
+
+            # update the units
+            cube.units = "mm hr-1"
```

Comment:
```suggestion
            # Convert from kg m-2 s-1 to mm s-1 assuming 1kg water = 1l water = 1dm^3 water.
            # This is a 1:1 conversion, so we just change the units.
            cube.units = "mm s-1"
            # Convert the units to per hour.
            cube.convert_units("mm hr-1")
```
Per Kirsty's suggestion we can use convert_units if we first set the units to mm $s^{-1}$.